anndata/_core/sparse_dataset.py

```diff
@@ -304,10 +300,17 @@ def _normalize_index(
             row, col = np.ix_(row, col)
         return row, col
 
+    # def __setitem__(self, index: Union[Index, Tuple[()]], value):
+
+    #     row, col = self._normalize_index(index)
+    #     mock_matrix = self._to_backed()
+    #     mock_matrix[row, col] = value
+
```

Comment:
@ivirshup `__setitem__` does work in so far as the tests pass.  So I think it would be worth adding the `DeprecationWarning` (or perhaps some sort of `InstabilityWarning` which probably better describes this) to the `__setitem__` method.  Does that work?

Reply:
I didn't have anything in mind specifically, but I also thought we were not planning on actually deprecating this since I imagine eventually we do want to bring it back somehow.  So the instability would be referring to behavior surrounding performance and/or write location (i.e., if we ever move to `kerchunk`).

Reply:
I think deprecation would be fine.

Even if there were atomic updates via `kerchunk` (what I think you're talking about?) this would require changes in user code to setup.